lib/msf/core/auxiliary/epmp.rb

```diff
@@ -152,20 +153,23 @@ def login_2(user, pass, epmp_ver)
         )
 
         # get the cookie now
-        sysauth_value_2 = res.headers['Set-Cookie'].match(/((.*)[$ ])/)
-        stok_value_2_dirty = res.body.match(/"stok": "(.*?)"/)
-        stok_value_2 = "#{stok_value_2_dirty}".split('"')[3]
-        final_cookie = "#{sysauth_value_2}" + 'usernameType_80=admin; stok_80=' + "#{stok_value_2}"
+        cookies = res.get_cookies_parsed
+        stok_value_dirty = res.body.match(/"stok": "(.*?)"/)
+        stok_value = "#{stok_value_dirty}".split('"')[3]
+        sysauth_dirty = cookies.values.select { |v| v.to_s =~ /sysauth_/ }.first.to_s
+        sysauth_value = sysauth_dirty.match(/((.*)[$ ])/)
+
+        final_cookie = "#{sysauth_value}" + 'usernameType_80=admin; stok_80=' + "#{stok_value}"
 
         # create config_uri for different modules
-        config_uri_dump_config = '/cgi-bin/luci/;stok=' + "#{stok_value_2}" + '/admin/config_export?opts=json'
-        config_uri_reset_pass = '/cgi-bin/luci/;stok=' + "#{stok_value_2}" + '/admin/set_param'
-        config_uri_get_chart = '/cgi-bin/luci/;stok=' + "#{stok_value_2}" + '/admin/get_chart'
+        config_uri_dump_config = '/cgi-bin/luci/;stok=' + "#{stok_value}" + '/admin/config_export?opts=json'
```

Comment:
here and the next 2 lines, there is no need to interpolate `stok_value`, you can use it literally with `+`

Reply:
Done